sdk/program/src/bignum.rs

```diff
@@ -0,0 +1,850 @@
+//! @brief Solana Rust-based BigNumber for bpf-programs
+use std::fmt;
+
+#[repr(C)]
+#[derive(Debug)]
+/// Structure for passing BigNumbers to syscalls
+pub struct FfiBigNumber {
+    pub data: u64,
+    pub data_len: u64,
+    pub is_negative: bool,
+}
```

Comment:
These fields need comments, but as I understand it data is a pointer, data_len says how many `u64`s data has.

I'm not sure how this would interact with rust, but would it be nicer to have `data_len` and `is_negative` as first fields, and then data as a variable-length array. That would save a malloc and keep it in a single struct.

Reply:
Changed in 2289d94cbed61cf1250360cef0556a87e15c07cc

Reply:
data is still a pointer rather than an variable length array of `u8`.

Reply:
After giving this some thought, I don't think this makes much sense. Maybe just keep the data as a pointer rather than a variable length array.